File: tests/fixtures/adapter-parity/get_user.py
# get_user.py — User service module.

from dataclasses import dataclass


@dataclass
class User:
    id: int
    name: str
    email: str


def get_user(id: str) -> User:
    if not id:
        raise ValueError("get user: id is empty. Provide a valid user ID.")
    return User(id=1, name="Alice", email="alice@example.com")


def create_user(name: str, email: str) -> User:
    if not name:
        raise ValueError("create user: name is empty. Provide a name.")
    return User(id=2, name=name, email=email)

File: tests/fixtures/adapter-parity/GetUser.java
// GetUser.java — User service class.

package service;

public class UserService {
    public record User(int id, String name, String email) {}

    public User getUser(String id) {
        if (id == null || id.isEmpty()) {
            throw new IllegalArgumentException("get user: id is empty. Provide a valid user ID.");
        }
        return new User(1, "Alice", "alice@example.com");
    }

    public User createUser(String name, String email) {
        if (name == null || name.isEmpty()) {
            throw new IllegalArgumentException("create user: name is empty. Provide a name.");
        }
        return new User(2, name, email);
    }
}
